homework_31/src/main/java/org/web_services/OrderController.java

```diff
@@ -0,0 +1,41 @@
+package org.web_services;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.web_services.entity.Order;
+
+import java.util.List;
+
+@RequestMapping("api/orders")
+@RestController
+public class OrderController {
+    private OrderRepository orderRepository;
+
+    public OrderController(OrderRepository orderRepository) {
+        this.orderRepository = orderRepository;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Order> getOrderById(@PathVariable int id) {
+        Order order = orderRepository.getOrderById(id);
+        if (order != null) {
+            return ResponseEntity.ok(order);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
```

Comment:
The controller should not contain any if-else, try-catch, etc. Would be better to throw an exception on the service level (you should write one) + implement your custom exception handler. Its a best practice and a recommended way to go.